### PR TITLE
Adjust output defaults

### LINKED ISSUE #118: brimcap analyze: unexpected pretty print behavior
The stats output for brimcap has two different modes: 1) status line and 2) json

Currently it is doing some unintuitive behavior when choosing which status behavior to choose:
- If input is a terminal and brimcap analyze output is going to a terminal, brimcap runs in status line mode (it should actually have no stats output)
- If input is a terminal and brimcap analyze output is not a terminal (e.g. a file), brimcap runs in json mode (it should run in stats line mode)

### PR DESCRIPTION
**Pull request by mattnibs**
Change the heuristic enabling json logging by default: If stdin, stdout and
stderr are all not terminals.

analyze stats output: If pretty status line output is enabled display
stats iff:
- -o is a file: display stats
- -o is stdout and stdout is NOT a terminal: display stats

Closes #118

### CODE REVIEW

File: cmd/brimcap/root/command.go [hunk 1]
```diff
@@ -53,7 +53,9 @@ type Command struct {
 func New(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
 	c := &Command{}
 	c.cli.SetFlags(f)
-	isterm := term.IsTerminal(int(os.Stdout.Fd()))
+	isterm := term.IsTerminal(int(os.Stdout.Fd())) ||
+		term.IsTerminal(int(os.Stdin.Fd())) ||
+		term.IsTerminal(int(os.Stderr.Fd()))
```
Comment (nwt) [inline cmd/brimcap/root/command.go:58]: `-json` only affects stderr, so I wouldn't expect stdin or stdout to influence it.
```suggestion
	isterm := term.IsTerminal(int(os.Stderr.Fd()))
```

File: cmd/brimcap/analyze/command.go [hunk 2]
```diff
@@ -46,16 +48,24 @@ type Command struct {
...
+	f.BoolVar(&c.nostats, "nostats", false, "do not display stats in stderr")
 	return c, nil
 }
 
+// json:
+// - always display stats in stderr (except if -nostats is enabled)
+// status line display stats iff:
+// - -o is a file: display stats
+// - -o is stdout and stdout is NOT a terminal: display stats
+
```
Comment (nwt) [inline cmd/brimcap/analyze/command.go:68]: Put this adjacent to the code it describes so it'll get updated if that code changes.

Comment (mattnibs) [reply cmd/brimcap/analyze/command.go:68]: Sure!

File: cmd/brimcap/root/command.go (hunk 1 above)
Comment (mattnibs) [reply cmd/brimcap/root/command.go:58]: Makes sense... I think...

**Review by nwt — APPROVED** [review]
:shipit:

**Pull request merged**